portal/src/graphql/portal/AnonymousUsersConfigurationScreen.tsx

```diff
@@ -84,6 +90,10 @@ const AnonymousUserConfigurationScreen: React.FC = function AnonymousUserConfigu
 
   const [state, setState] = useState(constructStateFromAppConfig(appConfig));
 
+  useEffect(() => {
+    setState(constructStateFromAppConfig(appConfig));
+  }, [appConfig]);
```

Comment:
Can you explain more on this fix? This fix looks weird to me. In what case `appConfig` is changed?

Reply:
Before appConfig is loaded (receive response in query), appConfig is null, so it changes from null to the initial appConfig in response.

Reply:
The screen code now looks like

```typescript
function MyScreen() {
  const { loading, error, refetch, data } = useAppConfigQuery(appID);
  if (loading) { return <Loading /> }
  if (error) { return <Error error={error} retry={refetch} /> }
  // use data, even if data may be null.
}
```

I suggest we change ALL screens (not this screen only) to the following structure.

```typescript
function MyScreen() {
  const { loading, error, refetch, data } = useAppConfigQuery(appID);
  if (loading) { return <Loading /> }
  if (error) { return <Error error={error} retry={refetch} /> }
  // data should not be null now. MyScreenContent does not need to use useEffect to refetch.
  return <MyScreenContent data={data} />
}
```